lib/Index/Index.cpp

```diff
@@ -334,34 +334,47 @@ class IndexSwiftASTWalker : public SourceEntityWalker {
     // get label locations
     auto *MemberwiseInit = DeclRef->getDecl();
     std::vector<SourceLoc> LabelLocs;
+    ArrayRef<Identifier> Labels;
     auto NameLoc = DeclRef->getNameLoc();
     if (NameLoc.isCompound()) {
       size_t LabelIndex = 0;
       SourceLoc ArgLoc;
       while ((ArgLoc = NameLoc.getArgumentLabelLoc(LabelIndex++)).isValid()) {
         LabelLocs.push_back(ArgLoc);
       }
+      Labels = MemberwiseInit->getFullName().getArgumentNames();
     } else if (auto *CallParent = dyn_cast_or_null<CallExpr>(getParentExpr())) {
       LabelLocs = CallParent->getArgumentLabelLocs();
+      Labels = CallParent->getArgumentLabels();
     }
 
     if (LabelLocs.empty())
       return;
 
+    assert(Labels.size() == LabelLocs.size());
```

Comment:
This is "obviously correct" for the call case, but given that the compound name case is checking isValid() and skipping, I'm a bit suspicious.  Could you explain how we know this is true?